src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -4444,11 +4442,21 @@ class StubGenerator: public StubCodeGenerator {
     __ movptr(state, state_mem);
 #endif
     __ movptr(subkeyHtbl, subkeyH_mem);
-    __ movptr(avx512_subkeyHtbl, avx512_subkeyH_mem);
     __ movptr(counter, counter_mem);
+// Save rbp and rsp
+    __ push(rbp);
+    __ movq(rbp, rsp);
+// Align stack
+    __ andq(rsp, -64);
+    __ subptr(rsp, 96 * longSize); // Create space on the stack for htbl entries
+    __ movptr(avx512_subkeyHtbl, rsp);
 
     __ aesgcm_encrypt(in, len, ct, out, key, state, subkeyHtbl, avx512_subkeyHtbl, counter);
 
+    __ addptr(rsp, 96 * longSize);
```

Comment:
I don't think you need this instruction since you restore `RSP` in the next.
Otherwise looks good.
Testing passed fine.